dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/dataintegrity/DefaultDataIntegrityService.java

```diff
@@ -751,6 +774,20 @@ public FlattenedDataIntegrityReport getFlattenedDataIntegrityReport()
         return groupVariablesByProgram( ruleVariables );
     }
 
+    private String getInvalidExpressionDescription( String expression )
+    {
+        try
+        {
+            expressionService.getIndicatorExpressionDescription( expression );
+        }
+        catch ( org.hisp.dhis.parser.expression.ParserException e )
+        {
+           return e.getMessage();
+        }
+
+        return "";

```

Comment:
Better to return null when valid, and ignore later?

Reply:
Code refactored.
this method will always end up in exception as it is fetching description of all invalid expressions.
I am working on returning some meaning full message instead of an exception message.

Reply:
Good. Merging now (just raise new PR for improvements).